src/store/CategoricalDisplayStore.js

```diff
@@ -127,24 +127,33 @@ export default class CategoricalDisplayStore {
     }
 
     @computed get filteredItems() {
-        let filteredItems = toJS(this.getCategoricalItems()),
-            items = (filteredItems && filteredItems[0]) ? filteredItems[0].data : [];
+        let filteredItems = toJS(this.getCategoricalItems());
 
         if (this.favoritesId) {
             const favsCategory = toJS(this.favoritesCategory);
-            let favsCategoryItem = [];
-
-            favsCategory.data.forEach( favItem => {
-                if ( typeof favItem === 'string') {
-                    let itemObj = items.find( item => item.itemId === favItem);
-                    if (itemObj) {
-                        favsCategoryItem.push(itemObj);
-                    }
+            let favsCategoryItem = favsCategory.data.filter( favItem => (typeof favItem !== 'string') );
+            const findFavItem = category =>{
```

Comment:
@shayanypn make this a member function that returns and array.
And the do

```js
filtereredItems.forEach(category => {
     const foundItems = findFavItem(category);
     favsCateogryItem.push(...foundItems);
})
```

As a rule of thumb, try to keep utility functions pure